feat: add namespace and sequencer signature verification to block execution outputs

tosynthegeek opened the pull request:
This PR addresses two related issues to the block execution program:

1. **Add namespace to block execution program outputs** (#97)
2. **Add sequencer signature verification** (#95)

## Changes Made

### Namespace Addition (#97)
- Added `namespace: Namespace` field to `BlockExecOutput` struct
- Updated output creation in main execution program to include the namespace that was verified
- The namespace is already available as input and used for verification, now it's also committed as output

### Sequencer Signature Verification (#95)
- Added `public_key: Vec<u8>` field to `BlockExecOutput` struct  
- Added public key as input to the SP1 program via `sp1_zkvm::io::read_vec()`
- Implemented signature verification logic that:
  - Validates the signer's public key in `SignedData` matches the expected sequencer public key
  - Verifies the signature over the protobuf-encoded `Data` using Ed25519
- Added `ed25519-dalek = "2"` dependency for signature verification
- Both public key validation and signature verification occur in the blob equivalency section

Closes #97
Closes #95

damiannolan reviewed (approved):
Thanks! ❤️ 

I can regen some real proofs later on for the testdata directory.

tosynthegeek commented:
> Thanks! ❤️ I addressed comments and follow ups.
> 
> I can regen some real proofs later on for the testdata directory.

Sure ser. Would also look to pick up another issue😊
